units/util.go

```diff
@@ -23,17 +24,15 @@ func HandlePoisonedHost(ctx context.Context, env evergreen.Environment, h *host.
 			if err != nil {
 				return errors.Wrap(err, "error getting containers")
 			}
-			catcher := grip.NewBasicCatcher()
+
 			for _, container := range containers {
 				catcher.Add(DisableAndNotifyPoisonedHost(ctx, env, container, reason))
 			}
 			catcher.Add(DisableAndNotifyPoisonedHost(ctx, env, *parent, reason))
-			if catcher.HasErrors() {
-				return catcher.Resolve()
-			}
 		}
 	}
-	return DisableAndNotifyPoisonedHost(ctx, env, *h, reason)
+	catcher.Add(DisableAndNotifyPoisonedHost(ctx, env, *h, reason))
```

Comment:
Does this still disable the host twice, one at 29 and once here?

Reply:
Yep, added an explicit check to noop those